optimum/habana/transformers/generation/utils.py

```diff
@@ -408,12 +518,15 @@ def generate(
             )
 
         # 5. Prepare `input_ids` which will be used for auto-regressive generation
+
         if self.config.is_encoder_decoder:
-            input_ids = self._prepare_decoder_input_ids_for_generation(
-                batch_size,
+            input_ids, _ = self._prepare_decoder_input_ids_for_generation(
+                batch_size=batch_size,
+                model_input_name=model_input_name,
+                model_kwargs=model_kwargs,
                 decoder_start_token_id=generation_config.decoder_start_token_id,
                 bos_token_id=generation_config.bos_token_id,
-                model_kwargs=model_kwargs,
+                max_length=generation_config.max_length or generation_config.max_new_tokens,
```

Comment:
Hmm I have a doubt here. I think `max_length` will be updated according to the value of `max_new_tokens`. I'm going to check.

Reply:
@regisss  max_length will be updated but that happens after this part of the code. [https://github.com/huggingface/optimum-habana/pull/274/files/6147008361d20bce0883ec1989d658ddb039f2a9#diff-284d6c109e6788a4405e302113203f6b7e98be68afc7dd3d85b89c6329b9275eR553]